packages/compass-crud/src/stores/crud-store.ts

```diff
@@ -1584,6 +1600,72 @@ class CrudStoreImpl
   openCreateSearchIndexModal() {
     this.localAppRegistry.emit('open-create-search-index-modal');
   }
+
+  openBulkDeleteDialog() {
+    const PREVIEW_DOCS = 5;
+
+    this.setState({
+      bulkDelete: {
+        previews: this.state.docs?.slice(0, PREVIEW_DOCS) || [],
+        status: 'open',
+        affected: this.state.count || 0,
+      },
+    });
+  }
+
+  bulkDeleteInProgress() {
+    this.setState({
+      bulkDelete: {
+        ...this.state.bulkDelete,
+        status: 'in-progress',
+      },
+    }); // TODO: COMPASS-7328
+  }
+
+  bulkDeleteFailed(ex: Error) {
+    return ex; // TODO: COMPASS-7328
+  }
+
+  bulkDeleteSuccess() {
+    this.closeBulkDeleteDialog();
+    // TODO: COMPASS-7328
+  }
+
+  closeBulkDeleteDialog() {
+    this.setState({
+      bulkDelete: {
+        ...this.state.bulkDelete,
+        status: 'closed',
+      },
+    });
+  }
+
+  async runBulkDelete() {
+    const { affected } = this.state.bulkDelete;
+    this.closeBulkDeleteDialog();
+
+    const confirmation = await showConfirmation({
+      title: 'Are you absolutely sure?',
+      buttonText: 'Delete',
+      description: `This action can not be undone. This will permanently delete ${
+        affected || 0
+      } documents.`,
+      variant: 'danger',
+    });
+
+    if (confirmation) {
+      this.bulkDeleteInProgress();
+      try {
+        await this.dataService.deleteMany(
+          this.state.ns,
+          this.state.query.filter
+        );
+        this.bulkDeleteSuccess();
```

Comment:
I am not sure how we want to handle the current list of documents that was rendered before delete. It still is visible on the screen - are we planning to refresh the list or leave it as it?

Reply:
its just that since there's no toast yet (which is todo for next ticket), it feels like nothing happened.

Reply:
That's a good point. We were discussing it yesterday. The issue with refreshing is that a delete can take minutes (or more) and if we refresh while the user is doing something else (we don't block the screen) there might be some weird behaviour. So we will leave it as it is and notify by using the toast, so the user can refresh.

The next task is the in-progress toast so there will be some feedback for the user at that point.